pkg/kubelet/cm/helpers.go

```diff
@@ -18,9 +18,58 @@ package cm
 
 import (
 	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
 	evictionapi "k8s.io/kubernetes/pkg/kubelet/eviction/api"
+	"sync"
 )
 
+// CGroupPods records the relationship between Pod UID and its cgroup name
+type CGroupPods struct {
```

Comment:
I don't really understand the reason for this code.  It appears to only be used in tests and the associated stubs, but the code logic that dictates the behavior in this pr is all oriented around changing the internal state of the pod manager cache.

Reply:
As you noted, this code is only for test.

Kubelet#HandlePodCleanups is the major method which checks whether it is safe to delete mirror pod.
I want to give related methods called by Kubelet#HandlePodCleanups (especially Kubelet#deleteOrphanedMirrorPods) good test coverage.

CGroupPods wraps the mapping between pod UID and cgroup name with a mutex since the test in TestDeleteOrphanedMirrorPods is multi-threaded. I spent some time to get rid of data races in the unit tests and came to this formation.

I am open to suggestion on the naming / location of such helper class.